components/expression/expression-renderer.js

```diff
@@ -172,21 +211,66 @@ genotet.ExpressionRenderer.prototype.drawExpressionMatrix_ = function() {
  * @private
  */
 genotet.ExpressionRenderer.prototype.drawMatrixCells_ = function() {
-  // TODO(liana): Implement this...
-  var cells = this.svgHeatmapContent_.selectAll('rect.cell').data([{
-    width: this.heatmapWidth_,
-    height: this.heatmapHeight_
-  }]);
-  cells.enter().append('rect')
-    .classed('cell', true);
+  var heatmapData = this.data.matrix;
+  var cellWidth = this.heatmapWidth_ / heatmapData.conditionNames.length;
+  var cellHeight = this.heatmapHeight_ / heatmapData.geneNames.length;
+  //console.log(heatmapData);
+  var colorScale = d3.scale.linear()
+    .domain([heatmapData.valueMin, heatmapData.valueMax])
+    .range(genotet.data.redYellowScale);
+
+  this.svgHeatmapContent_.selectAll('g').remove();
+  var cells = this.svgHeatmapContent_.selectAll('rect.cell').data(heatmapData.values);
+  var heatmapRows = cells.enter().append('g')
+    .attr("transform", genotet.utils.getTransform([
+      0,
+      this.conditionLabelHeight_ + this.profileHeight_
+    ]));
   cells.exit().remove();
-  cells
-    .attr('width', function(cell) {
-      return cell.width;
+  var heatmapRects = heatmapRows.selectAll('.rect')
+    .data(_.identity)
+    .enter().append('rect')
+    .attr('width', cellWidth)
```

Comment:
You are only setting properties on enter(). This works because you are re-rendering everything.
1) Please fix your rendering scheme so that it handles enter(), exit() and update properly.
2) Keep all properties NOT subject to change when re-rendered in enter(). Move those properties that would change to update.